opensearch-operator/pkg/builders/cluster_test.go

```diff
@@ -207,6 +208,22 @@ var _ = Describe("Builders", func() {
 		})
 	})
 
+	When("When Reconciling the cluster reconciler", func() {
+		It("should create a snapshotconfig batch job", func() {
+			clusterObject := ClusterDescWithVersion("2.2.1")
+
+			var snapshotRepoSettings = map[string]string{"bucket": "opensearch-s3-snapshot", "region": "us-east-1", "base_path": "os-snapshot"}
+			snapshotConfig := opsterv1.SnapshotRepoConfig{
+				Name:     "os-snap",
+				Type:     "s3",
+				Settings: snapshotRepoSettings,
+			}
+			clusterObject.Spec.General.SnapshotRepositories = []opsterv1.SnapshotRepoConfig{snapshotConfig}
+			result := NewSnapshotRepoconfigUpdateJob(&clusterObject, "snapshotrepoconfig", "foobar", "snapshotrepoconfig/checksum", nil, nil)
+			Expect(result.Spec.Template.Spec.Containers[0].Name).To(Equal("snapshotrepoconfig"))
```

Comment:
Please test more here, check that the configured repo is actually part of the cmd

Reply:
Added more unit tests.